pinot-broker/src/main/java/org/apache/pinot/broker/routing/HelixExternalViewBasedRouting.java

```diff
@@ -594,7 +594,7 @@ public String dumpSnapshot(String tableName)
     ArrayNode routingTableSnapshot = JsonUtils.newArrayNode();
 
     for (String currentTable : _routingTableBuilderMap.keySet()) {
-      if (tableName == null || currentTable.startsWith(tableName)) {
+      if (tableName == null || tableName.equals(TableNameBuilder.extractRawTableName(currentTable))) {
```

Comment:
By Pinot naming convention, `tableName` means table name with or without type suffix. If the table name provided has the suffix, we should not return the one with the different suffix.
`table1` will match `table12` is a great example why we should not use `startsWith`. To solve that, the check should be on the type. You can us `TableNameBuilder.getTableTypeFromTableName()` to get the type. If type is null, compare the raw table name; if type is not null, check if the name is the same.